File: fizzbuzz02.js

var fizzbuzzArray = [
    1,
    2,
    "fizz",
    4,
    "buzz",
    "fizz",
    7,
    8,
    "fizz",
    "buzz",
    11,
    "fizz",
    13,
    14,
    "fizzbuzz",
]

for(var cnt = 0; cnt < 100;) {
  for(var num = 0; num < fizzbuzzArray.length && cnt < 100; num++, cnt++) {
    process.stdout.write(fizzbuzzArray[num] + " ")
  }
  for(var num = 0; num < fizzbuzzArray.length; num++) {
    if(isNaN(fizzbuzzArray[num]) === false) {
      fizzbuzzArray[num] += 15
    }
  }
}

File: fizzbuzz02.go
package main

import "fmt"

func main() {
	var fizz int
	var buzz int
	for no := 1; no <= 100; no++ {
		fizz++
		buzz++
		if fizz == 3 && buzz == 5 {
			fmt.Print("fizzbuzz ")
			fizz = 0
			buzz = 0
		} else if fizz == 3 {
			fmt.Print("fizz ")
			fizz = 0
		} else if buzz == 5 {
			fmt.Print("buzz ")
			buzz = 0
		} else {
			fmt.Printf("%d ", no)
		}
	}
}
